src/widgets/pagination/__tests__/pagination-test.js

```diff
@@ -136,3 +136,71 @@ describe('pagination()', () => {
     };
   }
 });
+
+describe('pagination MaxPage', () => {
+  jsdom({useEach: true});
+
+  let ReactDOM;
+  let container;
+  let widget;
+  let results;
+  let helper;
+  let cssClasses;
+  let expectedProps;
+
+  beforeEach(() => {
+    ReactDOM = {render: sinon.spy()};
+    pagination.__Rewire__('ReactDOM', ReactDOM);
+    pagination.__Rewire__('autoHideContainerHOC', sinon.stub().returns(Pagination));
+
+    container = document.createElement('div');
+    cssClasses = {
+      root: 'root',
+      item: 'item',
+      link: 'link',
+      page: 'page',
+      previous: 'previous',
+      next: 'next',
+      first: 'first',
+      last: 'last',
+      active: 'active',
+      disabled: 'disabled'
+    };
+    results = {hits: [{first: 'hit', second: 'hit'}], nbHits: 300, hitsPerPage: 10, nbPages: 30};
+    helper = {
+      setCurrentPage: sinon.spy(),
+      search: sinon.spy()
+    };
+
+    expectedProps = {
+      createURL: function noRefCheck() {},
+      cssClasses: {active: 'ais-pagination--item__active active', disabled: 'ais-pagination--item__disabled disabled', first: 'ais-pagination--item__first first', item: 'ais-pagination--item item', last: 'ais-pagination--item__last last', link: 'ais-pagination--link link', next: 'ais-pagination--item__next next', page: 'ais-pagination--item__page page', previous: 'ais-pagination--item__previous previous', root: 'ais-pagination root'},
+      labels: {first: '«', last: '»', next: '›', previous: '‹'},
+      nbHits: 300,
+      nbPages: 30,
+      padding: 3,
+      setCurrentPage: function noRefCheck() {},
+      shouldAutoHideContainer: false,
+      showFirstLast: true
+    };
+  });
```

Comment:
All this boilerplate (css classes, labels, spies) makes the whole tests really hard to read and understand. Any change in the way we handle CSS will break the `maxPages` tests. I know this is not especially related to this issue, but we need to find a better way.

What about:

- Movign the simple if/else/min logic into its own `getMaxPage(maxPages, result.nbPages)` method, so we can test it separatly, with all the edge cases
- Mock this method (and making it return something like `42`), then check that `Pagination` is really called with this props.

Of course, it would be better if we could simply test that a component is called with a specific value in its props.